streams/src/main/java/org/apache/kafka/streams/errors/ProductionExceptionHandler.java

```diff
@@ -30,22 +30,60 @@ public interface ProductionExceptionHandler extends Configurable {
      *
      * @param record The record that failed to produce
      * @param exception The exception that occurred during production
+     * @deprecated Use {@link #handle(ErrorHandlerContext, ProducerRecord, Exception)} instead
      */
-    ProductionExceptionHandlerResponse handle(final ProducerRecord<byte[], byte[]> record,
-                                              final Exception exception);
+    @Deprecated
+    default ProductionExceptionHandlerResponse handle(final ProducerRecord<byte[], byte[]> record,
+                                                      final Exception exception) {
+        throw new UnsupportedOperationException();
```

Comment:
KIP says `NotImplementedException` -- we should update the KIP -- same for the other method below.

Reply:
KIP has been updated